# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: third_party/rust/wgpu-core/src/device/queue.rs
# Commit: b3109ad14685
# Full Hash: b3109ad14685624ada7861083ff5332aec9a7818
# Author: Andy Leiserson <aleiserson@mozilla.com>
# Date: 2025-10-07 17:00 +0000
# Description:
#   Bug 1992842 - Update wgpu to upstream 1072b878 (2025-10-06) r=webgpu-reviewers,supply-chain-reviewers,ErichDonGubler
# ==============================================================================

diff -r 1d3f000cf1f9 -r b3109ad14685 third_party/rust/wgpu-core/src/device/queue.rs
--- a/third_party/rust/wgpu-core/src/device/queue.rs	Tue Oct 07 16:51:44 2025 +0000
+++ b/third_party/rust/wgpu-core/src/device/queue.rs	Tue Oct 07 17:00:40 2025 +0000
@@ -183,9 +183,9 @@
                     fence.as_ref(),
                     last_successful_submission_index,
                     #[cfg(not(target_arch = "wasm32"))]
-                    timeout_ms,
+                    Some(core::time::Duration::from_millis(timeout_ms)),
                     #[cfg(target_arch = "wasm32")]
-                    0, // WebKit and Chromium don't support a non-0 timeout
+                    Some(core::time::Duration::ZERO), // WebKit and Chromium don't support a non-0 timeout
                 )
             };
             // Note: If we don't panic below we are in UB land (destroying resources while they are still in use by the GPU).